### PR TITLE
add v22.1.0-beta.3 release notes

### PR DESCRIPTION
**Pull request by taroface**
Drafted release notes for v22.1.0-beta.3.

@MichaelTrestman This release is slated for Monday the 18th, so would you be able to review this week?

### CODE REVIEW

File: _includes/releases/v22.1/v22.1.0-beta.3.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+- `LATEST` files are no longer overwritten and are now versioned and written in the `/metadata/latest` directory for non-mixed clusters. [#79553][#79553]
+- Previously, the [`ALTER CHANGEFEED`](../v22.1/alter-changefeed.html) statement would not work with changefeeds that use fully qualified names in their [`CREATE CHANGEFEED`](../v22.1/create-changefeed.html) statement. This is now fixed by ensuring that each existing target is added with its fully qualified name so that it can be resolved in validation checks. Every changefeed will now display the fully qualified name of every target in the [`SHOW CHANGEFEED JOB`](../v22.1/show-jobs.html) query. [#79745][#79745]
+- Added a `changefeed.backfill.scan_request_size` setting to control scan request size during backfill. [#79710][#79710]
+
+<h3 id="v22-1-0-beta-3-sql-language-changes">SQL language changes</h3>
+
+- CockroachDB now ensures the user passes the same number of locality-aware URIs for the full [backup](../v22.1/take-full-and-incremental-backups.html) destination as the `incremental_location` parameter (i.e., `BACKUP INTO LATEST IN ($1, $2, $3) WITH incremental_location = ($4, $5, $6)`). [#79600][#79600]
```
Comment (mikeCRL) [inline _includes/releases/v22.1/v22.1.0-beta.3.md:18]: Not a fan of i.e. and e.g. in docs. Here, it looks like it could be replaced with "for example".

File: _includes/releases/v22.1/v22.1.0-beta.3.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+- `EXPLAIN (DDL)`, when invoked on statements supported by the declarative schema changer, prints a plan of what the schema changer will do. This can be useful for anticipating the complexity of a schema change (e.g., anything involving backfill or validation operations might be slow to run) and for troubleshooting. `EXPLAIN (DDL, VERBOSE)` produces a more detailed plan. [#79780][#79780]
```
Comment (mikeCRL) [inline _includes/releases/v22.1/v22.1.0-beta.3.md:19]:
```suggestion
- `EXPLAIN (DDL)`, when invoked on statements supported by the declarative schema changer, prints a plan of what the schema changer will do. This can be useful for anticipating the complexity of a schema change (for example, anything involving backfill or validation operations might be slow to run) and for troubleshooting. `EXPLAIN (DDL, VERBOSE)` produces a more detailed plan. [#79780][#79780]
```

File: _includes/releases/v22.1/v22.1.0-beta.3.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+<h3 id="v22-1-0-beta-3-operational-changes">Operational changes</h3>
+
+- Added a new time-series metric, `storage.marked-for-compaction-files`, for the count of files marked for compaction. This is useful for monitoring storage-level background migrations. [#79370][#79370]
+- [Changefeed](../v22.1/change-data-capture-overview.html) creation and failure event logs are now emitted to the [`TELEMETRY` logging channel](../v22.1/eventlog.html#telemetry-events). [#79749][#79749]
```
Comment (mikeCRL) [inline _includes/releases/v22.1/v22.1.0-beta.3.md:24]:
```suggestion
- [Changefeed](../v22.1/change-data-capture-overview.html) creation and failure event logs are now emitted to the [`TELEMETRY`](../v22.1/eventlog.html#telemetry-events) logging channel. [#79749][#79749]
```

File: _includes/releases/v22.1/v22.1.0-beta.3.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+<h3 id="v22-1-0-beta-3-command-line-changes">Command-line changes</h3>
+
+- Introduced a new `ttllogger` [workload](../v22.1/cockroach-workload.html) which creates a TTL table emulating a "log" with rows expiring after the duration specified in the `--ttl` flag. [#	][#79482]
```
Comment (mikeCRL) [inline _includes/releases/v22.1/v22.1.0-beta.3.md:28]:
```suggestion
- Introduced a new `ttllogger` [workload](../v22.1/cockroach-workload.html) which creates a TTL table emulating a "log" with rows expiring after the duration specified in the `--ttl` flag. [#79482][#79482]
```

File: _includes/releases/v22.1/v22.1.0-beta.3.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+<h3 id="v22-1-0-beta-3-db-console-changes">DB Console changes</h3>
+
+- Filtering by column has been added to the [Hot Ranges page](../v22.1/ui-hot-ranges-page.html). [#79647][#79647]
```
Comment (mikeCRL) [inline _includes/releases/v22.1/v22.1.0-beta.3.md:32]:
```suggestion
- The [Hot Ranges page](../v22.1/ui-hot-ranges-page.html) now allows filtering by column. [#79647][#79647]
```

File: _includes/releases/v22.1/v22.1.0-beta.3.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+- Fixed a bug that caused an error when accessing a named column of a labeled tuple. The bug only occurred when an expression could produce one of several different tuples. For example, `(CASE WHEN true THEN (ROW(1) AS a) ELSE (ROW(2) AS a) END).a` would fail to evaluate. This bug was present since v22.1.0. Although present in previous versions, it was impossible to encounter due to limitations that prevented using tuples in this way. [#79529][#79529]
+- Previously, queries reading from an index or primary key on `FLOAT` or `REAL` columns `DESC` would read `-0` for every `+0` value stored in the index. This has been fixed to correctly read `+0` for `+0` and `-0` for `-0`. [#79533][#79533]
+- Separate tenants now use unique values for the internal field `ClusterID`. Previously, separate tenants that were hosted on the same storage cluster would be assigned the same `ClusterID`. [#79742][#79742]
```
Comment (mikeCRL) [inline _includes/releases/v22.1/v22.1.0-beta.3.md:40]:
```suggestion
```

Drop because we are no longer including multi-tenant architecture features in release notes.

File: _includes/releases/v22.1/v22.1.0-beta.3.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+- Fixed some cases where a job or schema change that had encountered an error would continue to execute for some time before eventually failing. [#79713][#79713]
+- Previously, the optional `is_called` parameter of the `setval` function would default to `false` when not specified. It now defaults to `true` to match PostgreSQL behavior. [#79779][#79779]
+- Updated props on the [Raft Messages](../v22.1/ui-debug-pages.html) page to include functions to set the time window and time scale in order to fix the date picker and drag-to-zoom functionality. [#79791][#79791]
```
Comment (mikeCRL) [inline _includes/releases/v22.1/v22.1.0-beta.3.md:43]:
```suggestion
- Updated props on the [Raft Messages](../v22.1/ui-debug-pages.html) page to include functions to set the time window and time scale in order to fix the date picker and drag-to-zoom functionality. [#79791][#79791]
```

I think the mention of props is internal, regarding how the page is built, and this alternative would be sufficient.

```suggestion
- On the [Raft Messages](../v22.1/ui-debug-pages.html) page, the date picker and drag-to-zoom functionality are now fixed. [#79791][#79791]
```

File: _includes/releases/v22.1/v22.1.0-beta.3.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+- Fixed a bug where Pebble compaction heuristics could allow a large compaction backlog to accumulate, eventually triggering high read amplfiication. [#79597][#79597]
```
Comment (mikeCRL) [inline _includes/releases/v22.1/v22.1.0-beta.3.md:44]:
```suggestion
- Fixed a bug where Pebble compaction heuristics could allow a large compaction backlog to accumulate, eventually leading to high read amplification. [#79597][#79597]
```

File: _includes/releases/v22.1/v22.1.0-beta.3.md [hunk 9]
```diff
@@ -0,0 +1,78 @@
+## v22.1.0-beta.3
+
+Release Date: April 18, 2022
+
+{% include releases/release-downloads-docker-image.md release=include.release %}
+
+<h3 id="v22-1-0-beta-3-enterprise-edition-changes">Enterprise edition changes</h3>
+
+- Unified the syntax for defining the behavior of initial scans on [changefeeds](../v22.1/create-changefeed.html) by extending the `initial_scan` option to accept three possible values: `'yes|no|only'`. [#79471][#79471]
```
[the remark this answers is not in the document]
Comment (kathancox) [reply _includes/releases/v22.1/v22.1.0-beta.3.md:9]: Yes that change is good. I have an anchor link on the initial scan option, so could change "changefeeds" to change-data-capture-overview.html and "`initial_scan`" to create-changefeeds.html#initial-scan.

File: _includes/releases/v22.1/v22.1.0-beta.3.md [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+- Changefeed messages for tables with multiple [column families](../v22.1/column-families.html) using the `split_column_families` option or the [`CREATE CHANGEFEED for TABLE bar FAMILY f_1`](../v22.1/create-changefeed.html) syntax will append the family name to the table name in the default topic. For example, `CREATE CHANGEFEED FOR TABLE foo, TABLE bar FAMILY f_1` will emit to the topics `foo` and `bar.f_1`.  In cloud storage sinks such as s3, the filename will include the family separated by a `+` (e.g., `bar+f_1`). A changefeed created with the `split_column_families` option will create new topics if new families are added to a watched table that already had multiple column families. [#79448][#79448]
```
[the remark this answers is not in the document]
Comment (kathancox) [reply _includes/releases/v22.1/v22.1.0-beta.3.md:10]: Yes, this looks good, but I actually think this could be shortened since the docs for the feature were merged last week? 

Changefeeds can now target tables with [more than one column family](use-changefeeds.html#changefeeds-on-tables-with-column-families) using either the [`split_column_families` option](create-changefeeds.html#split-column-families) or the `FAMILY` keyword . Changefeeds will emit individual messages per column family on a table.

Either way it looks good to me, here are some relevant links:
**The column family work**: use-changefeeds.html#changefeeds-on-tables-with-column-families, provides a description of the working & gotchas 
**example**: create-a-changefeed-on-a-table-with-column-families. 
**option anchor link**: create-changefeeds.html#split-column-families

**Review by taroface — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13604)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @MichaelTrestman and @mikeCRL)

___
*[_includes/releases/v22.1/v22.1.0-beta.3.md, line 9 at r2](https://reviewable.io/reviews/cockroachdb/docs/13604#-N-Zu67XpwlqbDGFUi2G-r2-9:-N-xTX_-6RXIE5sP4zbs:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c7b8b76c277495c5a6295f909a2c03243eeb7d01/_includes/releases/v22.1/v22.1.0-beta.3.md#L9)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Yes that change is good. I have an anchor link on the initial scan option, so could change "changefeeds" to change-data-capture-overview.html and "`initial_scan`" to create-changefeeds.html#initial-scan.
</blockquote></details>

Done.
___
*[_includes/releases/v22.1/v22.1.0-beta.3.md, line 12 at r2](https://reviewable.io/reviews/cockroachdb/docs/13604#-N-Zu67XpwlqbDGFUi2G-r2-12:-N-xVfarDjOWUe_GxyCC:byecx4s) ([raw file](https://github.com/cockroachdb/docs/blob/c7b8b76c277495c5a6295f909a2c03243eeb7d01/_includes/releases/v22.1/v22.1.0-beta.3.md#L12)):*
<details><summary><i>Previously, mikeCRL (Mike Lewis) wrote…</i></summary><blockquote>

I don't know and I don't think we define anywhere what a 'mixed cluster' is. I have a guess, but that's not too helpful. Can you rephrase or add to this?
</blockquote></details>

Thanks for pointing out. Looks like it means "mixed-version", so I've amended.
___
*[_includes/releases/v22.1/v22.1.0-beta.3.md, line 14 at r2](https://reviewable.io/reviews/cockroachdb/docs/13604#-N-Zu67XpwlqbDGFUi2G-r2-14:-N-xVqoDEs7sk6UJcoij:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c7b8b76c277495c5a6295f909a2c03243eeb7d01/_includes/releases/v22.1/v22.1.0-beta.3.md#L14)):*
<details><summary><i>Previously, mikeCRL (Mike Lewis) wrote…</i></summary><blockquote>

```suggestion
- Added a `changefeed.backfill.scan_request_size` setting to control scan request size during [backfill](../v22.1/use-changefeeds.html#schema-changes-with-column-backfill). [#79710][#79710]
```
</blockquote></details>

Done.
___
*[_includes/releases/v22.1/v22.1.0-beta.3.md, line 18 at r2](https://reviewable.io/reviews/cockroachdb/docs/13604#-N-Zu67XpwlqbDGFUi2G-r2-18:-N-xVt6w8rtmnJ186gak:b801efm) ([raw file](https://github.com/cockroachdb/docs/blob/c7b8b76c277495c5a6295f909a2c03243eeb7d01/_includes/releases/v22.1/v22.1.0-beta.3.md#L18)):*
<details><summary><i>Previously, mikeCRL (Mike Lewis) wrote…</i></summary><blockquote>

Not a fan of i.e. and e.g. in docs. Here, it looks like it could be replaced with "for example".
</blockquote></details>

Replaced. I've encountered and also used these infrequently. Is this worth adding to style guide?
___
*[_includes/releases/v22.1/v22.1.0-beta.3.md, line 19 at r2](https://reviewable.io/reviews/cockroachdb/docs/13604#-N-Zu67XpwlqbDGFUi2G-r2-19:-N-xW0grBDWFnqKq4SKv:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c7b8b76c277495c5a6295f909a2c03243eeb7d01/_includes/releases/v22.1/v22.1.0-beta.3.md#L19)):*
<details><summary><i>Previously, mikeCRL (Mike Lewis) wrote…</i></summary><blockquote>

```suggestion
- `EXPLAIN (DDL)`, when invoked on statements supported by the declarative schema changer, prints a plan of what the schema changer will do. This can be useful for anticipating the complexity of a schema change (for example, anything involving backfill or validation operations might be slow to run) and for troubleshooting. `EXPLAIN (DDL, VERBOSE)` produces a more detailed plan. [#79780][#79780]
```
</blockquote></details>

Done.
___
*[_includes/releases/v22.1/v22.1.0-beta.3.md, line 24 at r2](https://reviewable.io/reviews/cockroachdb/docs/13604#-N-Zu67XpwlqbDGFUi2G-r2-24:-N-xW3ib7GC4LfG6xD0c:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c7b8b76c277495c5a6295f909a2c03243eeb7d01/_includes/releases/v22.1/v22.1.0-beta.3.md#L24)):*
<details><summary><i>Previously, mikeCRL (Mike Lewis) wrote…</i></summary><blockquote>

```suggestion
- [Changefeed](../v22.1/change-data-capture-overview.html) creation and failure event logs are now emitted to the [`TELEMETRY`](../v22.1/eventlog.html#telemetry-events) logging channel. [#79749][#79749]
```
</blockquote></details>

Done.
___
*[_includes/releases/v22.1/v22.1.0-beta.3.md, line 28 at r2](https://reviewable.io/reviews/cockroachdb/docs/13604#-N-Zu67XpwlqbDGFUi2G-r2-28:-N-xW5Vg8aWDC8q5ozmJ:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c7b8b76c277495c5a6295f909a2c03243eeb7d01/_includes/releases/v22.1/v22.1.0-beta.3.md#L28)):*
<details><summary><i>Previously, mikeCRL (Mike Lewis) wrote…</i></summary><blockquote>

```suggestion
- Introduced a new `ttllogger` [workload](../v22.1/cockroach-workload.html) which creates a TTL table emulating a "log" with rows expiring after the duration specified in the `--ttl` flag. [#79482][#79482]
```
</blockquote></details>

Done.
___
*[_includes/releases/v22.1/v22.1.0-beta.3.md, line 32 at r2](https://reviewable.io/reviews/cockroachdb/docs/13604#-N-Zu67XpwlqbDGFUi2G-r2-32:-N-xW7Q-A0L7VfL8P3Zg:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c7b8b76c277495c5a6295f909a2c03243eeb7d01/_includes/releases/v22.1/v22.1.0-beta.3.md#L32)):*
<details><summary><i>Previously, mikeCRL (Mike Lewis) wrote…</i></summary><blockquote>

```suggestion
- The [Hot Ranges page](../v22.1/ui-hot-ranges-page.html) now allows filtering by column. [#79647][#79647]
```
</blockquote></details>

Done.
___
*[_includes/releases/v22.1/v22.1.0-beta.3.md, line 40 at r2](https://reviewable.io/reviews/cockroachdb/docs/13604#-N-Zu67XpwlqbDGFUi2G-r2-40:-N-xW95ZDhtOsyKUWf77:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c7b8b76c277495c5a6295f909a2c03243eeb7d01/_includes/releases/v22.1/v22.1.0-beta.3.md#L40)):*
<details><summary><i>Previously, mikeCRL (Mike Lewis) wrote…</i></summary><blockquote>

```suggestion
```

Drop because we are no longer including multi-tenant architecture features in release notes.
</blockquote></details>

Done.
___
*[_includes/releases/v22.1/v22.1.0-beta.3.md, line 43 at r2](https://reviewable.io/reviews/cockroachdb/docs/13604#-N-Zu67XpwlqbDGFUi2G-r2-43:-N-xWDZ36XVLEzdC9NHx:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c7b8b76c277495c5a6295f909a2c03243eeb7d01/_includes/releases/v22.1/v22.1.0-beta.3.md#L43)):*
<details><summary><i>Previously, mikeCRL (Mike Lewis) wrote…</i></summary><blockquote>

```suggestion
- Updated props on the [Raft Messages](../v22.1/ui-debug-pages.html) page to include functions to set the time window and time scale in order to fix the date picker and drag-to-zoom functionality. [#79791][#79791]
```

I think the mention of props is internal, regarding how the page is built, and this alternative would be sufficient.

```suggestion
- On the [Raft Messages](../v22.1/ui-debug-pages.html) page, the date picker and drag-to-zoom functionality are now fixed. [#79791][#79791]
```
</blockquote></details>

Done.
___
*[_includes/releases/v22.1/v22.1.0-beta.3.md, line 44 at r2](https://reviewable.io/reviews/cockroachdb/docs/13604#-N-Zu67XpwlqbDGFUi2G-r2-44:-N-xWFRXAYqwooVrb6iF:b-u2hi7e) ([raw file](https://github.com/cockroachdb/docs/blob/c7b8b76c277495c5a6295f909a2c03243eeb7d01/_includes/releases/v22.1/v22.1.0-beta.3.md#L44)):*
> - Fixed a bug where Pebble compaction heuristics could allow a large compaction backlog to accumulate, eventually leading to high read amplification. [#79597][#79597]

Nice, fixed.
